raster-foundry/raster-foundry
# Add MultiTiff image rendering for more datasources

## Overview

This PR enables multitiff imagery rendering for more datasources.

- [x] Sentinel-2
- [x] Landsat 4 / 5
- [x] Landsat 7
- [ ] MODIS

It however doesn't work1 Landsat 4/5/7 it turns out aren't stored as COGs, so they'll always require some processing. MODIS also isn't stored as a COG.

However! This doesn't need to be catastrophic. We can change what upload processing / ingest means for Sentinel-2 / the other datasources so that we don't have to do quite as much work and we'll still get better import performance.

For Sentinel-2 this means gdal_translate to make the internal tiling scheme have smaller tiles. For the others, this means adding overviews at all. They already have internal tiling, but that's not enough --

```bash
$ gdalinfo /vsicurl/https://storage.googleapis.com/gcp-public-data-landsat/LT05/01/004/027/LT05_L1TP_004027_19850918_20170218_01_T1/LT05_L1TP_0040
27_19850918_20170218_01_T1_B1.TIF
Driver: GTiff/GeoTIFF
Files: /vsicurl/https://storage.googleapis.com/gcp-public-data-landsat/LT05/01/004/027/LT05_L1TP_004027_19850918_20170218_01_T1/LT05_L1TP_004027_19850918_20170218_01_T1_B1.TIF
       /vsicurl/https://storage.googleapis.com/gcp-public-data-landsat/LT05/01/004/027/LT05_L1TP_004027_19850918_20170218_01_T1/LT05_L1TP_004027_19850918_20170218_01_T1_MTL.txt
Size is 7961, 7261
Coordinate System is:
PROJCS["WGS 84 / UTM zone 21N",
    GEOGCS["WGS 84",
        DATUM["WGS_1984",
            SPHEROID["WGS 84",6378137,298.257223563,
                AUTHORITY["EPSG","7030"]],
            AUTHORITY["EPSG","6326"]],
        PRIMEM["Greenwich",0,
            AUTHORITY["EPSG","8901"]],
        UNIT["degree",0.0174532925199433,
            AUTHORITY["EPSG","9122"]],
        AUTHORITY["EPSG","4326"]],
    PROJECTION["Transverse_Mercator"],
    PARAMETER["latitude_of_origin",0],
    PARAMETER["central_meridian",-57],
    PARAMETER["scale_factor",0.9996],
    PARAMETER["false_easting",500000],
    PARAMETER["false_northing",0],
    UNIT["metre",1,
        AUTHORITY["EPSG","9001"]],
    AXIS["Easting",EAST],
    AXIS["Northing",NORTH],
    AUTHORITY["EPSG","32621"]]
Origin = (383085.000000000000000,5364015.000000000000000)
Pixel Size = (30.000000000000000,-30.000000000000000)
Metadata:
  AREA_OR_POINT=Point
  METADATATYPE=ODL
Image Structure Metadata:
  COMPRESSION=LZW
  INTERLEAVE=BAND
Corner Coordinates:
Upper Left  (  383085.000, 5364015.000) ( 58d34'48.59"W, 48d25' 6.36"N)
Lower Left  (  383085.000, 5146185.000) ( 58d31'21.18"W, 46d27'32.76"N)
Upper Right (  621915.000, 5364015.000) ( 55d21' 8.24"W, 48d25' 2.94"N)
Lower Right (  621915.000, 5146185.000) ( 55d24'44.50"W, 46d27'29.58"N)
Center      (  502500.000, 5255100.000) ( 56d58' 0.61"W, 47d26'57.62"N)
Band 1 Block=256x256 Type=Byte, ColorInterp=Gray
```

### Checklist


### Notes

Sentinel-2 imagery only works if it's one of the small slivers. I suspect this is due to a combination of the data being in eu-central-1 (so far away) and having a large internal tiling system (1024x1024) for some bands, which makes reading data more expensive. I don't know how to confirm this exactly though. It's possible it would be fine if we pushed histogram calculation down to gdal (see #5172) but we don't know that that's the case.

Acknowledging that it's broken in certain ways, I think it's still fine, since it's effectively feature flagged :man_shrugging:, and it's still good I think to test known ways that it's broken and get the feature flagged thing out

## Testing Instructions

- add a Landsat 4 /5 and a Landsat 7 image to a project from the CMR
- ~try to process it -- you should get an error from the API about missing overviews~ <-- that was true before I realized that I'd break upload processing everywhere. Instead, _it should be fine_, to prove that in reverting those changes I didn't break anything
- turn BACKSPLASH_ENABLE_MULTITIFF on in docker-compose for the api-server and backsplash
- start servers + frontend
- head to the stock sentinel project (http://localhost:9091/projects/edit/df697018-68ea-4b4e-81b3-50481a4bf15f/scenes?page=1) and zoom in on it so you can see that visualizing it works
- add a Landsat 8 unprocessed image to a project to confirm I didn't break that

Closes #4879

Closes #5181
